tests/test_data.py

```diff
@@ -1,21 +1,111 @@
+from access_nri_intake.data import CATALOG_NAME_FORMAT
+from access_nri_intake.data.utils import _get_catalog_rp, available_versions
 
 
-def test_get_catalog_fp():
+@mock.patch("access_nri_intake.data.utils.get_catalog_fp")
+def test__get_catalog_rp(mock_get_catalog_fp, test_data):
     """
-    Check that we're getting the correct path to the catalog.yaml file. We need
-    to ensure that this works both in editable & non-editable installs.
+    Check that we correctly decipher to rootpath (rp) to the catalogs
     """
-    INSTALL_DIR = Path(access_nri_intake.__file__).parent
-    expected_path = os.path.join(INSTALL_DIR, "data/catalog.yaml")
+    mock_get_catalog_fp.return_value = test_data / "catalog/catalog-good.yaml"
+    assert (
+        access_nri_intake.data.utils.get_catalog_fp()
+        == test_data / "catalog/catalog-good.yaml"
+    ), "Mock failed"
+
+    rp = _get_catalog_rp()
+    assert (
+        rp == "/this/is/root/path/"
+    ), f"Computed root path {rp} != expected value /this/is/root/path/"
+
+
+@mock.patch("access_nri_intake.data.utils.get_catalog_fp")
+@pytest.mark.parametrize(
+    "cat", ["catalog/catalog-bad-path.yaml", "catalog/catalog-bad-structure.yaml"]
+)
+def test__get_catalog_rp_runtime_errors(mock_get_catalog_fp, test_data, cat):
+    """
+    Check that we correctly decipher to rootpath (rp) to the catalogs
+    """
+    mock_get_catalog_fp.return_value = test_data / cat
+    assert (
+        access_nri_intake.data.utils.get_catalog_fp() == test_data / cat
+    ), "Mock failed"
+
+    with pytest.raises(RuntimeError):
+        _get_catalog_rp()
+
+
+@mock.patch("access_nri_intake.data.utils._get_catalog_rp")
+def test_available_versions(mock__get_catalog_rp, test_data):
+    mock__get_catalog_rp.return_value = test_data / "catalog/catalog-dirs"
+    cats = available_versions(pretty=False)
+    assert cats == [
+        "v2025-02-28",
+        "v2024-06-19",
+        "v2024-01-01",
+        "v2019-02-02",
+    ], "Did not get expected catalog list"
+
+
+@mock.patch("access_nri_intake.data.utils._get_catalog_rp")
+def test_available_versions_pretty(mock__get_catalog_rp, test_data, capfd):
+    mock__get_catalog_rp.return_value = test_data / "catalog/catalog-dirs"
+    available_versions(pretty=True)
+    captured, _ = capfd.readouterr()
+    assert (
+        captured == "v2025-02-28\nv2024-06-19\nv2024-01-01\nv2019-02-02(-->vN.N.N)\n"
+    ), "Did not get expected catalog printout"
+
+
+@pytest.mark.parametrize(
+    "name",
+    [
+        "v2024-01-12",
+        "v2024-02-18",
+        "v2024-03-07",
+        "v2024-04-22",
+        "v2024-05-14",
+        "v2024-06-19",
+        "v2024-07-18",
+        "v2024-08-01",
+        "v2024-09-17",
+        "v2024-10-11",
+        "v2024-11-13",
+        "v2024-12-29",
+        "v2024-02-31",  # Regex not strict enough to avoid slightly wrong dates
+        "v2024-04-31",  # Regex not strict enough to avoid slightly wrong dates
```

Comment:
I wonder if we might want to use strptime instead of a regex to parse the version in order to avoid letting through these slightly wrong dates? Probably unimportant

Reply:
If we really cared about having 'real' dates, we'd need a two-part validation - one to check that the version string starts with `v`, and then a second to check the date, possibly with `strptime`. I'm tempted to not bother given that the date really represents a version number, so it doesn't have to be a 'real' date, but I'm then tempted to make this change to avoid any confusion.

Reply:
The datetime module will let you stick prefixes on date strings, eg:

```python
>>> v_str = "v2024-10-13"
>>> datetime.strptime(v_str,"v%Y-%m-%d")
 datetime.datetime(2024, 10, 13, 0, 0)
>>> no_v_str = "2024-10-13"
>>> datetime.strptime(no_v_str ,"v%Y-%m-%d")
ValueError: time data '2024-10-13' does not match format 'v%Y-%m-%d'
```

But yeah, I'm inclined to agree that its unimportant - I wouldn't have even thought about it if the test wasn't there.